core/chains/evm/gas/l2_suggested_estimator.go

```diff
@@ -151,10 +151,13 @@ func (o *l2SuggestedEstimator) GetLegacyGas(_ []byte, l2GasLimit uint64, opts ..
 	if !ok {
 		return nil, 0, errors.New("estimator is not started")
 	}
+	if gasPrice != nil {
+		gasPrice = capGasPrice(gasPrice, maxGasPriceWei, o.config)
+	}
```

Comment:
Actually, the proper thing to do here would be to return an error. Optimism is a special case.

If we try to submit a transaction that is not priced high enough the node can get permanently stuck. So gas estimation should be an error if the estimator's value is higher than the max allowed.

Reply:
Great catch. Will throw an error here